src/exercises/cheriabi/Makefile

```diff
@@ -0,0 +1,25 @@
+.include "../common.mk"
+
+all: kern-read-over-baseline kern-read-over-cheri perm-vmem-baseline perm-vmem-cheri print-more-baseline print-more-cheri
+
+kern-read-over-baseline: kern-read-over.c
```

Comment:
You should really be able to do something like:

```
PROGBASES=	kern-read-over perm-vmem print-more

.include "../common.mk"
```
and have common.mk do:
```
.for progbase in ${PROGBASES}
.for variant in baseline cheri
CLEANFILES+=	${progbase}-${variant}

all: ${progbase}-${variant}

${progbase}-${variant}: ${progbase}.c
	${CC} -o ${.TARGET} ${.ALLSRC} ${CFLAGS} ${CFLAGS_${variant:tu}}
.endfor
.endfor

clean:
	rm -f ${CLEANFILES}
```

Then CFLAGS_BASELINE and CFLAGS_CHERI would need to contain the relevant flags.